src/domain/individualHarbor/forms/harborForm/utils/utils.ts

```diff
@@ -41,39 +38,31 @@ export const getHarbor = (harborData: HARBOR_FORM | undefined): Harbor | undefin
     zipCode,
     municipality: municipality || '',
     wwwUrl,
-    imageFile: getImageFile(imageFile),
-    maps: getMaps(maps),
+    existingImageFile: getImageFile(imageFile),
+    addedImageFile: undefined,
+    existingMaps: getMaps(maps),
+    addedMaps: [],
   };
 };
 
-type MapFiles = {
-  addMapFiles: File[];
-  removeMapFiles: string[];
-};
-
 export const mapValuesToMutation = (harborId: string, values: Harbor) => {
-  const { name, streetAddress, zipCode, municipality, wwwUrl, imageFile, maps = [] } = values;
-
-  const { addMapFiles, removeMapFiles } = maps.reduce<MapFiles>(
-    (acc, map) => {
-      if (map.markedForDeletion) {
-        return {
-          ...acc,
-          removeMapFiles: acc.removeMapFiles.concat(map.id as string),
-        };
-      }
-
-      if (map.data !== undefined) {
-        return {
-          ...acc,
-          addMapFiles: acc.addMapFiles.concat(map.data),
-        };
-      }
+  const {
+    name,
+    streetAddress,
+    zipCode,
+    municipality,
+    wwwUrl,
+    addedImageFile,
+    existingMaps = [],
+    addedMaps = [],
+  } = values;
 
-      return acc;
-    },
-    { addMapFiles: [], removeMapFiles: [] }
-  );
+  const removeMapFiles = existingMaps.reduce<string[]>((acc, map) => {
```

Comment:
This can be written with good old `filter` like this:
```suggestion
  const removeMapFiles = existingMaps.filter((map) => map.markedForDeletion).map((map) => map.id);
```

Reply:
That's true, good suggestion!